src/System.Text.Json/tests/JsonReaderTests.cs

```diff
@@ -230,11 +238,23 @@ private static Json ReadJson(string jsonString)
             var jsonReader = new JsonReader(jsonString);
             var jsonObjectMain = new Object();
             var jsonMembersMain = new List<Members>();
+            var jsonArrayMain = new Array();
+            var jsonElementsMain = new List<Elements>();
+
+            if (jsonString.Trim().Substring(0, 1) == "[")
```

Comment:
1. Trim is not that cheap as you think - it does need to iterate so it might be useful to store this partial result somewhere.
2. To compare I'd use .StartsWith(Utf8CodeUnit) overload - this one is pretty bad due to utf16 utf8 mixing (although it's not that slow)